### PR TITLE
Emoji react

### PR DESCRIPTION
**Pull request by IrochkaZ**
**_Task:_** https: //github.com/sulemanof/js-lectures/blob/master/react/homework-2.md
**_Description:_**  implemented a JSON based emoticon search
emogi-gif

### CODE REVIEW

File: emoji/src/components/container/container.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import './container.css';
+import Form from '../form';
+import EmoContainer from '../emo-container';
+
+export default class Container extends Component {
+    render() {
+        const { data } = this.props;
+        return (
+            <div className='main'>
+                <Form cb={data[1]} />
```
Comment (sulemanof) [inline emoji/src/components/container/container.js]: what is `cb` ?
Please, never give such names in code. It becomes unreadable

File: emoji/src/components/emo-container/emo-container.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React, { Component, Fragment } from 'react';
+import './emo-container.css';
+import EmoItem from '../emo-item';
+
+export default class EmoContainer extends Component {
+
+    setLister() {
```
Comment (sulemanof) [inline emoji/src/components/emo-container/emo-container.js]: what is a benefit of making this a separate method?
I suppose it just could be inside the `render` method

File: emoji/src/components/emo-container/emo-container.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+            if (data.inputValue === false) {
+                return <EmoItem key={Math.random()} symbol={emo.symbol} title={emo.title} />
+            }
+        });
+    }
+
+    render() {
+        return (
+            <Fragment>
```
Comment (sulemanof) [inline emoji/src/components/emo-container/emo-container.js]: Fragment is excess here, we only use it compose more than 1 components
```suggestion
```

File: emoji/src/components/form/form.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
+import './form.css'
+
+export default class Form extends Component {
+
+    render() {
+        const { cb } = this.props;
```
Comment (sulemanof) [inline emoji/src/components/form/form.js]: Again, `cb` - I can't get what is this at brief look

File: emoji/src/components/app/app.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                return emoji.title;
+            }
+        });
+
+        this.setState({ sort });
+    }
+
+    render() {
+        return (
+            <Container data={[this.state, this.handleChange]} />
```
Comment (sulemanof) [inline emoji/src/components/app/app.js]: Never pass props like this, please =)
Your code become really unreadable!
I would do something like:

```suggestion
            <Container
              emojiList={emojiList}
              inputValue={this.state.inputValue}
              onChange={this.handleChange}/>
```

File: emoji/src/components/app/app.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import './app.css';
+
+import emojiList from '../../data/emojiList.json';
+
+export default class App extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            sort: emojiList,
```
Comment (sulemanof) [inline emoji/src/components/app/app.js]: Storing `emojiList` in the state is excessive!
And again and again and again: never name variables in a such way!! Your code become unclear!

File: emoji/src/components/container/container.js (hunk 1 above)
Comment (IrochkaZ) [reply emoji/src/components/container/container.js]: change cb name to searchField

File: emoji/src/components/emo-container/emo-container.js (hunk 2 above)
Comment (IrochkaZ) [reply emoji/src/components/emo-container/emo-container.js]: setLister()  removed from class to render method

File: emoji/src/components/emo-container/emo-container.js (hunk 3 above)
Comment (IrochkaZ) [reply emoji/src/components/emo-container/emo-container.js]: Fragment removed

File: emoji/src/components/app/app.js (hunk 5 above)
Comment (IrochkaZ) [reply emoji/src/components/app/app.js]: note noted!:)

**Pull request merged**